jpype/_jstring.py

```diff
@@ -37,14 +39,14 @@ def __new__(cls, *args, **kwargs):
 
 
 @_jcustomizer.JImplementationFor("java.lang.String")
-class _JStringProto(object):
-    def __add__(self, other):
-        return self.concat(other)
+class _JStringProto:
+    def __add__(self, other: str) -> str:
```

Comment:
Is this correct?   The type of ``__add__`` can take str or JString and will produce a JString.

```
>>> import jpype
>>> jpype.startJVM()
>>> r = jpype.JString("hello")
>>> print(type(r+"there"))
<java class 'java.lang.String'>
print(type(r+jpype.JString("there")))
<java class 'java.lang.String'>

```

Reply:
I think you are right. I fully inherited this from https://github.com/jpype-project/jpype/pull/1135/files#diff-7d1d060f99980e468a65aef1f60ef16f98e9088aac885eaad9b8c7500a12a863L5, which also exhibited this (it used ``typing.Text`` instead of ``str`` directly).

I would be interested to find out what precisely these proto type annotations do/change - it isn't clear to me how/where they are actually picked up. Do you happen to know?

Reply:
I haven't used typing.  It is hints for a lexer as  far as I am aware.  I think it would trigger a warning on an IDE, but as I don't use Python IDEs I can't really test this.   There are two types of hinting.  Abstract hinting indicates that something works like.  While concrete hinting means this type only.  JPype is very forgiving in that with converters it can take many types and the return types may also be polymorphic.  Thus I have not placed any type hinting on classes.  (Again because I dont know how they look on an IDE when they are wrong.)